app/controllers/admin/routes.py

```diff
@@ -407,6 +432,45 @@ def removeFromSession():
 
     return ""
 
+@admin_bp.route('/manageServiceLearning/imported/<courseID>', methods = ['POST', 'GET'])
+@admin_bp.route('/manageServiceLearning/imported/<courseID>', methods = ['POST'])
+def alterImportedCourse(courseID):
+    """
+    This route handles a GET and a POST request for the purpose of imported courses. 
+    The GET request provides preexisting information of an imported course in a modal. 
+    The POST request updates a specific imported course (course name, course abbreviation, 
+    hours earned on completion, list of instructors) in the database with new information 
+    coming from the imported courses modal. 
+    """
+    if request.method == 'GET':
+        try:
+            targetCourse = Course.get_by_id(courseID)
+            targetInstructors = CourseInstructor.select().where(CourseInstructor.course == targetCourse)
+            
+            courseData = model_to_dict(targetCourse, recurse=False)
+            serviceHours = (list(CourseParticipant.select()
+                                                  .where(CourseParticipant.course_id == targetCourse.id))[0]
+                                                  .hoursEarned)
+            
+            courseInstructors = [model_to_dict(instructor.user) for instructor in targetInstructors]
+            courseData['instructors'] = courseInstructors
+            courseData["hoursEarned"] = serviceHours
+
+            return jsonify(courseData)
+        
+        except Exception as e:
+            flash("Course not found or something else went wrong")
```

Comment:
You never use `e` for anything so why not just `except Exception`?  Also, if you are just checking whether or not a course exists you should do `except DoesNotExist` to catch the specific exception you are expecting. Doing blanket exception handling without printing the exception is not great and can lead to very annoying debugging down the line.

Also, for the flash message, just say "Course not found" or "Could not find Course".